react/components/UserAddress/index.js

```diff
@@ -16,12 +16,28 @@ const UserAddress = ({
   showCityAndState = false,
   showPostalCode = false,
   showPrefix = true,
+  showIfEmpty = false,
 }) => {
   const { orderForm } = addressQueryProp
-  const { shippingData } = orderForm || {}
   const handles = useCssHandles(CSS_HANDLES)
 
-  if (!orderForm || !shippingData || !shippingData.address) {
+  useEffect(() => {
+    window.addEventListener('locationUpdated', () => {
+      addressQueryProp.refetch()
+    })
+
+    return () => {
+      window.removeEventListener('locationUpdated', () => {
+        addressQueryProp.refetch()
```

Comment:
This will not work.

It should be something like this:

```js
const handleLocationUpdated = () => addressQueryProp.refetch()

window.addEventListener('locationUpdated', handleLocationUpdated)

return () => {
  window.removeEventListener('locationUpdated', handleLocationUpdated)
}
```

It should reference the same function